src/core/syscall_handler.rs

```diff
@@ -206,8 +211,9 @@ impl SyscallHandler for BusinessLogicSyscallHandler {
         vm: &VirtualMachine,
         syscall_ptr: Relocatable,
     ) -> Result<(), SyscallHandlerError> {
-        let SyscallRequest::EmitEvent(request) =
-            self._read_and_validate_syscall_request("emit_event", vm, syscall_ptr)?;
+        let SyscallRequest::EmitEvent(request) = self._read_and_validate_syscall_request("emit_event", vm, syscall_ptr)? else {
```

Comment:
`let-else` is a bit new for us :(

Reply:
Want me to change it to a `match` ?

Reply:
Changed it due to `let-else` not stable in current CI version